src/Utilities/TypeTraits.hpp

```diff
@@ -1360,6 +1360,30 @@ template <typename T>
 using remove_cvref_wrap_t = typename remove_cvref_wrap<T>::type;
 // @}
 
+/// \ingroup TypeTraitsGroup
+/// \brief Permits designating a fundamental type for a container
+///
+/// \details Uses SFINAE to designate `get_fundamental_type<T>::type` as `T`
+/// when `T` itself is an appropriate fundamental type, and the contained type
+/// of a container which specifies a `value_type`
+///
+/// `get_fundamental_type_t<T>` is provided as a type alias to `typename
+/// get_fundamental_type_t<T>::type`
+template <typename T, typename Enable = std::nullptr_t>
```

Comment:
So what should happen if I have a container that does not have `::value_type`? If the answer is "It's undefined/ill-formed" then I think the correct way to implement this is with `void_t` rather than `Requires`. If it's not then this code is broken to begin with.

What about for containers like `unordered_map` where the `value_type` is a `pair`? How should that behave? Is that not a case covered?

Reply:
void_t sounds like a good solution for the second case, where I use a silly a==a test to make sure the value_type is there, but won't work for the first case, where I want to test for is_fundamental. I could use a `Requires<cpp:17::is_same_v<void_t<T::value_type>,void>>` for the second one, which is probably clearer.

Current behavior should be that if it can get to something that `is_fundamental_v<T>` via `::value_type`, the `type` in the struct is set to that, otherwise no type is set. `std::pair` doesn't have a `::value_type`, so an `unordered_map` would get no type field set. If this kind of usage is desired to return the `mapped_type`, I'd be happy to add more cases to the function.

Reply:
I have no issues saying this only works for `::value_type` containers (we can expand as necessary). In that case I would implement this as:

```cpp
template <typename T, typename cpp17::void_t<>>
struct get_fundamental_type {
  using type = T;
};
template <typename T>
struct get_fundamental_type<T, cpp17::void_t<typename T::value_type> {
  using type = typename get_fundamental_type<typename T::value_type>::type;
};
```
or, if you really want to do the fundamental check:
```cpp
template <typename T, typename cpp17::void_t<>>
struct get_fundamental_type {
  using type = tmpl::conditional_t<cpp17::is_fundamental_v<T>, T, NoSuchType>;
};
```

Reply:
nice! I hadn't known about conditional_t. That'll do nicely. :+1:

Reply:
Great! There's also a `std::conditional_t` but the `tmpl` one is slightly faster to compile :)